src/flamenco/runtime/fd_borrowed_account.h

```diff
@@ -43,6 +46,26 @@ typedef struct fd_borrowed_account fd_borrowed_account_t;
 
 #define FD_BORROWED_ACCOUNT_DECL(_x)  fd_borrowed_account_t _x[1]; fd_borrowed_account_init(_x);
 
+/* This macro provides the same scoping guarantees as the Agave client's 
+   borrowed account semantics. It allows for implict/explicit dropping of 
+   borrowed accounts' write locks. It's usage mirrors the use of
+   FD_SCRATCH_SCOPE_{BEGIN,END} */
+#define FD_BORROWED_ACCOUNT_TRY_BORROW_IDX( _ctx, _idx, _account ) do {   \
+  fd_borrowed_account_t * _account = NULL;                                \
+  int _err = fd_instr_borrowed_account_view_idx( _ctx, _idx, &_account ); \
+  if( FD_UNLIKELY( _err != FD_ACC_MGR_SUCCESS ) ) {                       \
+    return FD_EXECUTOR_INSTR_ERR_MISSING_ACC;                             \
+  }                                                                       \
+  int _acquire_result = fd_borrowed_account_acquire_write(_account);      \
```

Comment:
In some cases Agave doesn't check that the lock has been successfully acquired: https://github.com/firedancer-io/firedancer/blob/aa52e6043af790ea615919d789dbf091ebd9f21b/src/flamenco/runtime/program/fd_address_lookup_table_program.c#L148-L167

We should perhaps have an equivalent `FD_BORROWED_ACCOUNT_TRY_BORROW_IDX_UNCHECKED` macro for these cases.

Reply:
not ready for review

Reply:
I am not implementing this in this PR because we don't need to care as much about scoping if we aren't doing a check